### ISSUE TITLE
ExamReview #7

### ISSUE THREAD

**Issue by rayshroom** [seneca-btp500f19/btp500-f19#17]
7. Can you binary search a sorted linked list? Explain why or why not?

A: No. Since you cannot access the middle element in constant time.

**Comment by Silvyre** [comment]
It's perhaps also worth mentioning [this](https://cathyatseneca.gitbook.io/data-strutures-and-algorithms/definitions/binary-search-trees), from this course note:
>[Binary Search Trees] provide a way for us to do a "binary search" on a linked structure which is not possible with a linear linked list.

**Comment by musik8** [comment]
I read online that it is possible, just no real point. 
By not being able to access the middle element in constant time makes it an invalid binary search? 
Because it would take longer than O(log n) with a linked list?

**Comment by cathyatseneca** [comment]
a linked list is not the same as a binary search tree.  You can't bin search (in any meaningful sense) a linked list

**Comment by saketepe** [comment]
Yes, it is possible but not as fast, and it takes alot of changes to the structure of linked list that isn't really worth it.

**Comment by Silvyre** [comment]
> 
> 
> Yes, it is possible

> 
> 
> I read online that it is possible

According to [the definition of *binary search* within our course notes](https://cathyatseneca.gitbook.io/data-strutures-and-algorithms/algorithms-analysis/analysis-of-binary-search), this is incorrect.

Our course notes state that:
>  A binary search can only be done to lists if two criteria are true:
> 1. the list must be sorted
> 1. access to any element given its position must have a constant run time.   

Linked lists are unable to fulfill this second condition, so, therefore, it is not possible to binary search a linked list.
